src/Microsoft.ML/Models/ClusterMetrics.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Models
+{
+    /// <summary>
+    /// This class contains the overall metrics computed by cluster evaluators.
+    /// </summary>
+    public sealed class ClusterMetrics
+    {
+        private ClusterMetrics()
+        {
+        }
+
+        internal static List<ClusterMetrics> FromOverallMetrics(IHostEnvironment env, IDataView overallMetrics)
+        {
+            Contracts.AssertValue(env);
+            env.AssertValue(overallMetrics);
+
+            var metricsEnumerable = overallMetrics.AsEnumerable<SerializationClass>(env, true, ignoreMissingColumns: true);
+            if (!metricsEnumerable.GetEnumerator().MoveNext())
+            {
+                throw env.Except("The overall ClusteringMetrics didn't have any rows.");
+            }
+
+            var metrics = new List<ClusterMetrics>();
+            foreach (var metric in metricsEnumerable)
+            {
+                metrics.Add(new ClusterMetrics()
+                {
+                    AvgMinScore = metric.AvgMinScore,
+                    Nmi = metric.Nmi,
+                    Dbi = metric.Dbi,
+                });
+            }
+
+            return metrics;
+        }
+
+        /// <summary>
+        /// Davies-Bouldin Index.
+        /// </summary>
+        /// <remarks>
+        /// DBI is a measure of the how much scatter is in the cluster and the cluster separation.
+        /// </remarks>
+        public double Dbi { get; private set; }
+
+        /// <summary>
+        /// Normalized Mutual Information
+        /// </summary>
+        /// <remarks>
+        /// NMI is a measure of the mutual dependence of the variables.
+        /// Normalized variants work on data that already has cluster labels.
+        /// It returns values from 0 to 1, where higher numbers are better.
+        /// </remarks>
+        public double Nmi { get; private set; }
+
+        /// <summary>
+        /// Average minimum score.
+        /// </summary>
+        /// <remarks>
+        /// This makes sense for K-Means algorithm, where the 'score' is the distance from the centroid to the example.
```

Comment:
Can we rephrase the definition to
```
/// AvgMinScore is the average squared-distance of examples from the respective cluster centroids.
/// It is defined as 
/// AvgMinScore  = (1/m) * sum ((xi - c(xi))^2)
/// where m is the number of instances in the dataset.
/// xi is the ith instance and c(xi) is the centriod of the predicted cluster for xi.
```

Reply:
Thank you Zeeshan!

---
In reply to: [193260519](https://github.com/dotnet/machinelearning/pull/316#discussion_r193260519) [](ancestors = 193260519)